sencha-workspace/packages/slate-cbl/src/view/student/Dashboard.js

```diff
@@ -47,60 +47,95 @@ Ext.define('Slate.cbl.view.student.Dashboard', {
         '</li>',
         '</tpl>'
     ],
-
+    
     skillsTpl: [
         '<tpl for=".">',
-        '<li class="cbl-skill">',
-        '<h5 class="cbl-skill-name" data-descriptor="{Descriptor}" data-statement="{Statement}">{Descriptor}</h5>',
-        '<ul class="cbl-skill-demos">',
-        '<tpl for="this.getDemonstrationBlocks(values)">',
-        '<li class="cbl-skill-demo <tpl if="Level==0"> cbl-grid-demo-missed</tpl>" <tpl if="SkillID">data-skill="{SkillID}"</tpl> <tpl if="DemonstrationID">data-demonstration="{DemonstrationID}"</tpl>>',
-        '<tpl if="Level &gt;= 0">',
-        '{[values.Level == 0 ? "M" : values.Level]}',
-        '<tpl else>',
-        '&nbsp;',
-        '</tpl>',
-        '</li>',
-        '</tpl>',
-        '</ul>',
-        '<div class="cbl-skill-description"><p>{Statement}</p></div>',
+            '<li class="cbl-skill">',
+                '<h5 class="cbl-skill-name" data-descriptor="{Descriptor}" data-statement="{Statement}">{Descriptor}</h5>',
+                '<ul class="cbl-skill-demos" data-skill="{ID}">',
+                    '<tpl for="this.getDemonstrationBlocks(values)">',
+                        '<li class="cbl-skill-demo <tpl if="values.Level==0"> cbl-grid-demo-missed</tpl>" <tpl if="DemonstrationID">data-demonstration="{DemonstrationID}"</tpl>>',
+                            '<tpl if="values.Level &gt;= 0">',
+                                '{[values.Level == 0 ? "M" : values.Level]}',
+                            '<tpl else>',
+                                '&nbsp;',
+                            '</tpl>',
+                        '</li>',
+                    '</tpl>',
+                '</ul>',
+                '<div class="cbl-skill-description"><p>{Statement}</p></div>',
 //                '<div class="cbl-skill-complete-indicator cbl-level-{parent.level} is-checked">',
 //                    '<svg class="check-mark-image" width="16" height="16">',
 //                        '<polygon class="check-mark" points="13.824,2.043 5.869,9.997 1.975,6.104 0,8.079 5.922,14.001 15.852,4.07"/>',
 //                    '</svg>',
 //                '</div>',
-        '</li>',
+            '</li>',
         '</tpl>',
-        {
-            getDemonstrationBlocks: function(skill, studentId) {
-                var demonstrationsRequired = skill.DemonstrationsRequired,
-                    blocks = Ext.isArray(skill.demonstrations) ? skill.demonstrations : [];
-
-                blocks = Ext.Array.sort(blocks, function(a, b) {
-                    return a.Level < b.Level ? -1 : 1;
+        
+        {    sortDemonstrations: function sortDemonstrations(demonstrations, limit) {
+                'use strict';
+                demonstrations = Ext.isArray(demonstrations) ? demonstrations : [];
+                
+                var len = demonstrations.length,
+                    x = 0,
+                    i = 0,
```

Comment:
once there's more than one `for` loop in the function body and the iterators are being reused i move the initialization to the `for` opening and just declare them at the top of the function, so let's get rid of the assignments here